lua/wikis/commons/Infobox/Extension/TeamHistoryAuto.lua

```diff
@@ -0,0 +1,350 @@
+---
+-- @Liquipedia
+-- page=Module:Infobox/Extension/TeamHistoryAuto
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+--remaining issue: player joins/leaves same team with multiple roles on different dates (e.g. MarioMe LoL wiki)
+
+local Class = require('Module:Class')
+local Lua = require('Module:Lua')
+
+local Array = Lua.import('Module:Array')
+local DateExt = Lua.import('Module:Date/Ext')
+local FnUtil = Lua.import('Module:FnUtil')
+local Info = Lua.import('Module:Info', {loadData = true})
+local Json = Lua.import('Module:Json')
+local Logic = Lua.import('Module:Logic')
+local Namespace = Lua.import('Module:Namespace')
+local Table = Lua.import('Module:Table')
+local TransferModel = Lua.import('Module:Transfer/Model')
+local TransferRef = Lua.import('Module:Transfer/References')
+
+local Link = Lua.import('Module:Widget/Basic/Link')
+local HtmlWidgets = Lua.import('Module:Widget/Html/All')
+local Abbr = HtmlWidgets.Abbr
+local Fragment = HtmlWidgets.Fragment
+local Span = HtmlWidgets.Span
+local Tbl = HtmlWidgets.Table
+local Td = HtmlWidgets.Td
+local Th = HtmlWidgets.Th
+local Tr = HtmlWidgets.Tr
+local WidgetUtil = Lua.import('Module:Widget/Util')
+
+local Condition = require('Module:Condition')
+local ConditionTree = Condition.Tree
+local ConditionNode = Condition.Node
+local Comparator = Condition.Comparator
+local BooleanOperator = Condition.BooleanOperator
+local ColumnName = Condition.ColumnName
+
+local SPECIAL_ROLES = {
+	'Retired',
+	'Retirement',
+	'Military',
+	'Banned',
+	'Producer',
+	'Caster',
+	'Admin',
+	'Observer',
+	'Host',
+	'Talent',
+	'League Operator',
+	'Inactive'
+}
+local LOAN = 'Loan'
+local ROLE_CONVERT = Lua.import('Module:Infobox/Extension/TeamHistoryAuto/RoleConvertData', {loadData = true})
+
+local POSITION_ICON_DATA = Lua.requireIfExists('Module:PositionIcon/data', {loadData = true})
```

Comment:
Same

Reply:
this is the same we already use in `Module:TransferRow/Display`
additionally that module holds icon stuff while the new role data doesn't cover that (yet)
(it likely needs more wiki overwrites of the data too)

imho we should solve those together (and oos for this PR)